### PR TITLE
[Merged by Bors] - chore(CategoryTheory/Adjunction): address porting notes in `Reflective` file

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Adjunction/Unique.lean [hunk 1]
```diff
@@ -20,6 +20,13 @@ Everything else is deduced from this:
 
 * `Adjunction.rightAdjointUniq` : If `G` and `G'` are both right adjoint to `F`, then they are
   naturally isomorphic.
+
+## TODO
+
+There some overlap with the file `Adjunction.Mates`. In particular, `natTransEquiv` is just a
+special case of `mateEquiv`. However, before removing `natTransEquiv`, in favour of `mateEquiv`,
+the latter needs some more API lemmas such as `natTransEquiv_apply_app`, `natTransEquiv_id`, etc.
+in order to make automation work better in the rest of this file.
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Adjunction/Unique.lean:29]:
```suggestion
```

**Comment by dagurtomas** [comment]
> ⚠️ The sha of the head commit of this PR conflicts with #16086. Mergify cannot evaluate rules on this PR. ⚠️

What do I do with this information?

File: Mathlib/CategoryTheory/Adjunction/Reflective.lean [hunk 2]
```diff
@@ -153,60 +153,21 @@ instance [Reflective i] (X : Functor.EssImageSubcategory i) :
...
-
-lemma equivEssImageOfReflective_map_counitIso_app_inv [Reflective i]
-    (X : Functor.EssImageSubcategory i) :
-  (Functor.essImageInclusion i).map (equivEssImageOfReflective_counitIso_app X).inv =
-    (NatTrans.app (reflectorAdjunction i).unit X.obj) := rfl
+-- These attributes are necessary to make automation work in `equivEssImageOfReflective`.
+-- Making them global doesn't break anything elsewhere, but this is enough for now.
+-- TODO: investigate further.
+attribute [local simp 900] Functor.essImageInclusion_map
+attribute [local ext] Functor.essImage_ext
```
[the remark this answers is not in the document]
Comment (dagurtomas) [reply Mathlib/CategoryTheory/Adjunction/Reflective.lean]: Done

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**